src/event.rs

```diff
@@ -105,7 +103,7 @@ fn error_listeners() -> &'static ErrorListeners {
 }
 
 /// Listen to balance changes.
-pub fn on_balance_change<F: Fn(BalanceEvent) + Send + 'static>(cb: F) {
+pub fn on_balance_change<F: Fn(&BalanceEvent<'_>) + Send + 'static>(cb: F) {
     let mut l = balance_listeners()
```

Comment:
Do we really need to use Mutex for the listeners in the non async functions?

Reply:
I'm wrapping the listeners in a Mutex because the listeners are stored in a mutable vector (but it could also be a RwLock)

Reply:
Yes I mean do we need it? Because on_balance_change here it is not an async function and will be executed sequentially. So would like to know if there is a scenario that there are more than 1 location the listeners are read/written.

Reply:
(simultaneously I mean)

Reply:
it's always sequentially but it might be accessed by multiple threads on consumers of the library. but we mainly use it to make the Vec<Listeners> mutable.

Reply:
Got it. If multiple threads need to access the listeners in the same time then we need `Mutex` (or RwLock) here. LGTM.